### DISCUSSION TITLE
Address field layout inconsistency

### DISCUSSION THREAD

**Discussion by passatgt** [woocommerce/woocommerce#45050] [WooCommerce, Blocks, and Site Editing]
Hi! I just noticed that there are some inconsistency on how the form fields are rendered with the checkout block. To give you some examples:

With Hungary, notice theres two empty spaces:

<img width="763" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/43415/8fcc8d7a-21fc-482b-a164-ef6eae3d4ff5">
(also with Hungary, we could hide the state field, since it is not used anymore for shipping or billing, we don't need to collect this info, but this is another topic)

With Germany, the address field jumps to the top:

<img width="776" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/43415/1b7f8814-e09f-484b-b659-1f68e59f6c55">

With Japan, the layout looks okay, however if you check the country locale, the postcode field should be on the left, and the state field on the right:

<img width="755" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/43415/1b94e661-ade3-4db0-a840-ec7920a42e20">
<img width="465" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/43415/9ca5b8da-10fe-4557-bc98-13adb45c4a18">

The form-row-first and form-row-last classes are ignored in the checkout block. However, only Japan and Hungary actually uses these classes when defining the locale. 

On the old checkout, both Hungary and Japan looks good, because the country field is 100% wide:

<img width="579" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/43415/20225d0a-408f-4393-a23e-12ae71d3172e">

<img width="549" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/43415/4e76efd4-459d-4d32-b03c-4dff3fc57b0f">

And on the old checkout, street address was below the country field:

<img width="555" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/43415/ced47e0f-e247-48f8-ae29-f18fea7b9854">

Not sure about the solution, but something to think about. Both Hungary and Japan could be fixed in the locale to match the other countries, but the empty space would still be there and it would mess with the layout of the old checkout, since both are coming from the same function.

**Comment by senadir** [comment]
Thank you for starting this @passatgt, I think it's time to possibility reconsider how those fields flow and how they should handle empty space.

The Hungary/Japan issue is valid and I will convert this to an issue so we can work on a solve.
